docs/hostname.md

```diff
@@ -0,0 +1,41 @@
+---
+sidebar_label: Customize hostname
+title: ''
+---
+
+<head>
+  <link rel="canonical" href="https://elemental.docs.rancher.com/hostname"/>
+</head>
+
+### How to set a custom hostname
+
+When booting a host with an Elemental ISO, the hostname is temporarily set to `rancher-{$RANDOM}`.
+As soon as the boot process is finished, the registration phase takes place: the host connects to the
+Elemental Operator, which creates a MachineInventory for the host.
+
+Each host registered with the Elemental Operator is tracked by a `MachineInventory` resource.
+The hostname of the host is eventually set to the `name` of the `MachineInventory` resource,
+during the Cluster provisioning phase (i.e., during k3s/RKE2 deployment on the host).
+
+#### Default hostname
+
+The default name assigned to the newly created MachineInventory is in the form `m-{$UUID}`.
+When the host is provisioned as part of a Cluster, the `m-{UUID}` hostname is set, overriding the previous `rancher-{$RANDOM}` hostname.
```

Comment:
Shouldn't this be different for the next release?
I thought `rancher-{$RANDOM}` was removed with a recent PR.

I also miss the reference on how not to set the default hostname (and using the DHCP provided one).